coral/src/domain/helper/documentation-helper.ts

```diff
@@ -27,20 +27,41 @@ type MarkdownStringBrand = string & {
 
 type MarkdownString = Awaited<ReturnType<typeof createMarkdown>>;
 
+// very unique string to be able to identify documentation
+// transformation errors that otherwise can't be determined
+const documentationTransformError =
+  "6e6c85af-71b4-4021-b7fe-bf99c81f2c6aZ-d3841163-ba74-4db4-b874-35b973d2a80e-32e866bd-99d0-4a64-ac15-5b73104ab0ff";
+
+/** There can be edge cases where transforming `stringifiedHtml` from backend
+ * into a markdown string causes an error. To be able to handle that error
+ * gracefully while still getting `topicOverview` but also show our users
+ * helpful information, we set the value of `documentationTransformError`
+ * as property for topicOverview.topicDocumentation. If topicDocumentation
+ * matches this string, we know that there was an error and can inform the user.
+ */
+function documentationTransformationError(documentation: MarkdownStringBrand) {
+  return documentation === documentationTransformError;
+}
+
 // `createMarkdown` is used to transform documentation on the api response
 //  for example on TopicOverview. The `topicDocumentation` is a html string
 //  that we transform to markdown.
 async function createMarkdown(stringifiedHtml: string) {
   // see: https://unifiedjs.com/learn/recipe/remark-html/#how-to-turn-html-into-markdown
   // we're also using rehypeSanitize to sanitize the HTML
   if (unified) {
-    const result = await unified()
-      .use(rehypeParse)
-      .use(rehypeSanitize)
-      .use(rehypeRemark)
-      .use(remarkStringify)
-      .process(stringifiedHtml);
-    return String(result) as MarkdownStringBrand;
+    try {
+      const result = await unified()
+        .use(rehypeParse)
+        .use(rehypeSanitize)
+        .use(rehypeRemark)
+        .use(remarkStringify)
+        .process(stringifiedHtml);
+      return String(result) as MarkdownStringBrand;
+    } catch (error) {
+      console.error(error);
+      return documentationTransformError as MarkdownStringBrand;
```

Comment:
> Maybe we could consider not calling createMarkdown in the transformTopicOverviewResponse but in TopicDocumentation, instead? Then we could gracefully throw there without blocking the rest of the Topic overview in a more straightforward way?

We could, but the whole idea behind that was to keep all information that we're transforming html to markdown etc. in `domain`, so there's never a html string floating around outside of it. This aims to minimise the risk of passing that string anywhere without it being transformed (and sanitised etc pp).  And later when we switch to only markdown string, changing that is really straingforward to us. 

I would prefer to keep that a bit confusing error handling - that is covering a extremely unlikely edge case (*) - in favour of having the whole transformation process "behind bars". The risk of using the html string anywhere where it shouldn't go (which is already very low) is imho still higher then that error I had coming up in the client. 

(*) I don't even see how it could happen other then with the super weird problem I had locally. So the error handling is almost unnecessary, but I thought it's still worth adding it, because maybe someone developing in coral could run into it and then it's already helping a bit, because it narrows down where the error happened and what caused it.

Reply:
Hmm ok. It's true that it's a very specific edge case that is even hard to reproduce when you are trying :o

And it's true that it's good to have that transform happen independently of the endpoint user.

I still think it's a bit of a magic trick, and I'm not comfortable with it haha! But ok ^^